gidgethub/abc.py

```diff
@@ -32,12 +34,15 @@ def __init__(
         oauth_token: Opt[str] = None,
         cache: Opt[CACHE_TYPE] = None,
         base_url: str = sansio.DOMAIN,
+        rate_limiter=None,
```

Comment:
Better to pass in to the constructor or do via subclassing?

Reply:
I think it is better to inject it by passing to the constructor rather than by sub-classing so you do not end up with the outer product of (rate_limiters x I/O implementations) sub-classes.